# <center>Teste 🅰🅱️ - Incluses<center>

# **Introdução ao Teste AB**

Realizamos um teste AB para avaliar o desempenho de duas interfaces distintas, A e B, em um aplicativo voltado ao público-alvo de menores aprendizes e efetuados, com faixa etária entre 15 e 35 anos. Para isso, distribuímos a experiência entre duas amostras, com 28 pessoas usando a interface A e 26 pessoas usando a interface B. 

## **Coleta de dados**

A coleta de dados foi realizada de forma presencial, utilizando a ferramenta Maze para registrar o tempo que cada participante levava para completar cada fluxo. Durante esse processo, caso algum participante tivesse dificuldade em compreender as instruções, reforçávamos as orientações conforme necessário, garantindo que pudessem seguir adiante corretamente, mas em nenhum momento enviezamos o teste.

## **Objetivos**
O objetivo é entender qual interface promove uma experiência mais intuitiva e satisfatória para o público jovem e em início de carreira, explorando fluxos importantes de uso no aplicativo. Também queremos entender se a performace da interface B é melhor que a interface A por motivos de mudanças na interface ou motivos externos. O teste AB permitirá avaliar como essas mudanças impactam a performance, facilidade de uso e satisfação dos usuários em relação aos fluxos principais do aplicativo, fornecendo dados concretos para entender qual das interfaces oferece a melhor experiência para o público-alvo.

--- 

### **Hipóteses**

H0 : μB >= μA

H1 : μB < μA

--- 

## **Fluxos Avaliados**
Os fluxos críticos avaliados para verificar o impacto das mudanças entre as interfaces foram:

- Cadastro
- Login
- Inscrição em Vagas
- Logout
- Criação de Cursos

--- 

## **Mudanças entre as Interfaces**

#### 1. Fluxo de Cadastro e Login
**Interface A:** Botão de cadastro com borda posicionado embaixo e botão de login em cima.

**Interface B:** Botão de cadastro sem borda posicionado em cima e botão de login embaixo.

**Motivo da Mudança:** Esta modificação foi realizada para investigar se a posição e estilo dos botões influenciam a facilidade de uso ao logar ou se cadastrar. A escolha da Interface B é baseada em estudos de usabilidade de outros aplicativos, onde o botão de login costuma ter destaque, já que é mais utilizado.

#### 2. Fluxo de Inscrição em Vagas
**Interface A:** Informações da vaga apresentadas sem rótulos e botão de inscrição posicionado mais abaixo.

**Interface B:** Informações da vaga apresentadas com rótulos e botão de inscrição posicionado no centro.

**Motivo da Mudança:** Observamos que as informações da vaga não estavam suficientemente claras. Na Interface B, adicionamos rótulos às informações e alteramos a posição do botão de inscrição para avaliar se isso facilita o entendimento e melhora a performance do usuário.

#### 3. Fluxo de Logout
**Interface A:** Botão de logout acessível na barra de configurações do perfil do usuário.

**Interface B:** Botão de logout colocado diretamente no perfil do usuário.

**Motivo da Mudança:** Acreditamos que o botão de logout estava de difícil acesso. Alteramos a posição na Interface B para um local mais direto, visando melhorar a experiência de navegação ao facilitar o acesso ao logout.

#### 4. Fluxo de Criação de Curso
**Interface A:** Ícone de curso na barra de navegação representado por uma lâmpada. O botão de acesso aos cursos do usuário e criação é chamado de "Criar". As etapas de criação do curso parecem botões, mas não são interativas.

**Interface B:** Ícone de curso na barra de navegação representado por um livro. O botão de acesso aos cursos e criação é chamado de "Seus Cursos". As etapas de criação do curso não se assemelham a botões, melhorando a clareza.

**Motivo da Mudança:** Percebemos que o fluxo de criação de curso poderia ser mais intuitivo. A Interface B oferece uma navegação mais clara, com ícones e botões mais descritivos e o layout das etapas de criação menos confuso para evitar cliques incorretos.

# **Imports**

In [169]:
import pandas as pd
import pingouin as pg
import numpy as np
import stats

### Separando dados

In [170]:
# Lendo dataframes
df_interface_a = pd.read_excel('../bases/Interface_A.xlsx')
df_interface_b = pd.read_excel('../bases/Interface_B.xlsx')

display(df_interface_b)
display(df_interface_a)

Cadastro  Login  Inscrição da Vaga  Logout  Criação do Curso
0      20.40  18.08              23.95    4.76             38.11
1     152.88  10.66              11.61    7.50             28.05
2      33.75   3.24              22.24    8.66             12.25
3      92.81   7.99              12.85   25.06             25.41
4      33.15   3.67               9.78    6.03             24.05
5      44.09   4.19              12.08    6.24             15.70
6      44.09   4.19              12.08    6.24             15.70
7      14.95   4.67               5.09   19.44             66.98
8       9.95  14.46              11.76    9.55             30.44
9       6.32   3.85              11.07    6.44             17.86
10     28.13   7.25               7.93   15.06             35.92
11     46.30   3.85              13.63    4.05             33.73
12     62.18   3.98              17.44    6.13             14.04
13     23.49   8.95               9.98    6.13             19.19
14     32.55   3.48               6.29    5.05             21.92
15      4.53   4.36              13.30    6.27             19.98
16     29.01  16.59              10.15    5.21             22.44
17     21.10   6.65               7.05    5.55             30.32
18     14.77   5.67               4.82    5.65             17.42
19     26.21   7.06               5.09   10.61             22.63
20     23.57   4.47              10.07    4.00             43.22
21     37.69   4.15              10.53    8.67             22.45
22     35.04   6.17              10.87    5.49             21.57
23     49.98   6.78              16.66    3.18             21.27
24     25.32   3.44               2.88    5.59             11.52
25     24.55  11.49               3.79    4.69             20.39

Cadastro  Login  Inscrição da Vaga  Logout  Criação do Curso
0      28.83   5.85               9.65   10.46             22.00
1      10.89   6.44              15.92   11.79             24.61
2      52.08   6.70               9.31    8.88             17.34
3      53.26   4.83             200.59   10.41             43.01
4       4.63   4.27              14.56   17.43             21.43
5      17.14   7.77               4.52  199.61             15.85
6      45.45   6.92               9.84    6.59             17.03
7      37.68  11.58              21.84   34.38             27.52
8      33.40  13.56               6.23   11.56             37.84
9      26.20  16.09               8.50    6.35             25.15
10      6.34   5.46               4.88   13.00             22.93
11      7.42  12.00              23.15   24.29             40.99
12     22.26   4.72               5.28   14.01             13.35
13     12.71  25.84              17.88   15.37              6.24
14     14.36   3.59              13.47   12.43              8.14
15     13.53   3.07               4.19    8.33             24.69
16      5.77   7.11               4.81   10.06             27.44
17     20.36  64.12               8.13   16.16             16.86
18     15.21   4.06              10.66   68.64             18.61
19     29.17   5.94               5.38    9.46             32.30
20    121.68   3.27               9.21  152.81            127.26
21      4.48   3.04               4.35    8.90             70.62
22     32.38   6.47              10.93    8.21             26.50
23      5.40  20.64              12.52    9.34             94.52
24     27.84   8.83              19.95   45.96             32.32
25     58.00   4.30               4.34    5.71             12.01
26     10.82  21.02               9.58   11.05             15.56
27     28.84   5.25               9.66   10.47             23.00

### **Calculando a média de tempo por usuário**

In [171]:
# Interface A
medias_usuarios_a = []

for i in range(len(df_interface_a)):
    medias_usuarios_a.append(df_interface_a.loc[i].mean())

# Média final da Interface A
media_interface_a = sum(medias_usuarios_a) / len(medias_usuarios_a)

print("Média da interface A:", media_interface_a)
print("Tamanho da amostra para a interface A:", len(df_interface_a))

Média da interface A: 22.464142857142857
Tamanho da amostra para a interface A: 28


In [172]:
# Interface B
medias_usuarios_b = []

for i in range(len(df_interface_b)):
    medias_usuarios_b.append(df_interface_b.loc[i].mean())

# Média final da Interface B
media_interface_b = sum(medias_usuarios_b) / len(medias_usuarios_b)

print("Média da interface B:", media_interface_b)
print("Tamanho da amostra para a interface B:", len(df_interface_b))

Média da interface B: 17.330384615384617
Tamanho da amostra para a interface B: 26


## **Calculando teste T**

In [173]:
# Definindo o nível de significância
alpha = 0.05 

In [174]:
# Realizando o teste t com Pingouin
t_teste = pg.ttest(medias_usuarios_b, medias_usuarios_a, paired=False, alternative='less', correction='auto')
print(t_teste)

               T        dof alternative     p-val        CI95%  cohen-d BF10  \
T-test -1.510274  37.190986        less  0.069712  [-inf, 0.6]  0.40077  1.4   

          power  
T-test  0.42367  


### **Graus de liberdade**

In [175]:
gl = t_teste['dof'][0]
print("Grau de liberdade: ", gl)

# Interpretação
if t_teste['dof'][0] >= 30:
    print("O grau de liberdade é alto, indicando uma distribuição t mais próxima da normal.")
else:
    print("O grau de liberdade é baixo, o que implica uma maior variabilidade e incerteza.")

Grau de liberdade:  37.19098644416882
O grau de liberdade é alto, indicando uma distribuição t mais próxima da normal.


C:\Users\oliviadomingues-ieg\AppData\Local\Temp\ipykernel_15652\4160941899.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  gl = t_teste['dof'][0]
C:\Users\oliviadomingues-ieg\AppData\Local\Temp\ipykernel_15652\4160941899.py:5: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  if t_teste['dof'][0] >= 30:


#### **T-valor**

In [176]:
t_valor = t_teste['T'][0]
print("T-VALOR:", t_valor)

T-VALOR: -1.5102743769020166


C:\Users\oliviadomingues-ieg\AppData\Local\Temp\ipykernel_15652\1692808353.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  t_valor = t_teste['T'][0]


#### **P-valor**

In [177]:
p_valor = t_teste['p-val'][0]
print("P-VALOR:", p_valor)

P-VALOR: 0.06971190179384895


C:\Users\oliviadomingues-ieg\AppData\Local\Temp\ipykernel_15652\3365898475.py:1: FutureWarning: Series.__getitem__ treating keys as positions is deprecated. In a future version, integer keys will always be treated as labels (consistent with DataFrame behavior). To access a value by position, use `ser.iloc[pos]`
  p_valor = t_teste['p-val'][0]


#### **Interpretação do t-teste**

In [181]:
# Comparando o p-valor com o nível de significância
if p_valor <= alpha:
   
   print("Rejeitamos a hipótese nula. Os resultados indicam que a performance da interface A é significativamente maior que a da interface B. Essa diferença provavelmente se deve a fatores relacionados à estrutura da interface, embora outros fatores externos não possam ser completamente descartados.")

else:
    print("Aceitamos a hipótese nula (H0). Os resultados indicam que não há evidência suficiente para afirmar que a performance da interface A é significativamente menor que a da interface B. Por isso, podemos dizer que a interface B teve uma média menor de tempo por motivos externos e não por motivos de mudanças na interface.")

Aceitamos a hipótese nula (H0). Os resultados indicam que não há evidência suficiente para afirmar que a performance da interface A é significativamente menor que a da interface B. Por isso, podemos dizer que a interface B teve uma média menor de tempo por motivos externos e não por motivos de mudanças na interface.


### **T-crítico**

In [179]:
# Conservador
# Se os graus de liberdade não estão na tabela, então sempre arredondamos eles para baixo. Portanto, será 30
t_critico_conservador = -1.697

# Arriscado
# Sempre arredondamos para cima os graus de liberdade. Portanto, será 40
t_critico_arriscado = -1.684

In [180]:
# Interpretação

# Conservador
if t_valor < t_critico_conservador:
    print("Com base no t-crítico \033[1;35mconservador\033[0m, \033[1;31mrejeitamos a hipótese nula.\033[0m")
else:
    print("Com base no t-crítico \033[1;35mconservador\033[0m, \033[1;32maceitamos a hipótese nula.\033[0m")

# Arriscado
if t_valor < t_critico_arriscado:
    print("Com base no t-crítico \033[1;36marriscado\033[0m, \033[1;31mrejeitamos a hipótese nula.\033[0m")
else:
    print("Com base no t-crítico \033[1;36marriscado\033[0m, \033[1;32maceitamos a hipótese nula.\033[0m")

Com base no t-crítico conservador, aceitamos a hipótese nula.
Com base no t-crítico arriscado, aceitamos a hipótese nula.


# **Conclusão do Teste AB entre as Interfaces A e B**
Após a realização do teste AB entre as interfaces A e B, os resultados obtidos nos permitem fazer as seguintes considerações:

#### **Hipóteses**:
Testamos a hipótese nula (H0), que postula que a performance da interface A é igual ou superior à da interface B. A hipótese alternativa (H1) sugere que a performance da interface A é significativamente menor que a da interface B.

#### **Valores Críticos**:
Com um t-crítico conservador de -1.697 (graus de liberdade arredondados para baixo) e um t-crítico arriscado de -1.684 (graus de liberdade arredondados para cima), tanto os testes conservador quanto arriscado resultaram em uma aceitação da hipótese nula, já que o t-calculado de -1.510 está acima de ambos os valores críticos.
P-valor: O p-valor obtido foi de 0.0697, que está acima do nível de significância usual de 0.05. Isso indica que não há evidência estatística suficiente para rejeitar a hipótese nula.

#### **Grau de Liberdade**: 
Com um grau de liberdade calculado em aproximadamente 37.19, indicamos que as variâncias entre as amostras A e B são diferentes.

#### **Interpretação**:
Diante dos resultados, aceitamos a hipótese nula (H0). Isso significa que não encontramos evidências suficientes para afirmar que a performance da interface A é significativamente inferior à da interface B. Portanto, podemos concluir que, em termos de tempo de interação, a interface B não se destacou a ponto de comprovar que a interface A apresenta uma performance menor.

É importante ressaltar que a média menor de tempo na interface B pode ser atribuída a fatores externos e não necessariamente a melhorias nas mudanças feitas na interface. A ausência de diferença significativa entre as duas interfaces sugere que outras variáveis externas podem ter influenciado o desempenho dos usuários.